### PR TITLE
adding builtin function int2bin

### PR DESCRIPTION
**Pull request by UrbanskiDawid**
this is my first commit to this repo.
Please check it carefully.

Changes to runtime/object.go are up for discussion - motivation for this part is in 'padding' from spec (see below)

https://www.etsi.org/deliver/etsi_es/201800_201899/20187301/04.07.01_60/es_20187301v040701p.pdf

ETSI ES 201 873-1 V4.7.1 (2015-06)
page 288

> C.1.3 Integer to bitstring
int2bit(in integer invalue, in integer length) return bitstring
This function converts a single integer value to a single bitstring value. 
The resulting string is length bits long.
For the purposes of this conversion, a bitstring shall be interpreted as a positive base 2 integer value.
The rightmost bit is least significant, the leftmost bit is the most significant. 
The bits 0 and 1 represent the decimal values 0 and 1 respectively.
If the conversion yields a value with fewer bits than specified in the length parameter, then the bitstring shall be padded on the left with zeros.
In addition to the general error causes in clause 16.1.2, error causes are:
• invalue is less than zero;
• the conversion yields a return value with more bits than specified by length.

### CODE REVIEW

File: builtins/builtins.go [hunk 1]
```diff
@@ -31,6 +31,32 @@ func Rnd(args ...runtime.Object) runtime.Object {
...
+	lengthArg, lenOk := args[1].(runtime.Int)
+	if !lenOk {
+		return runtime.Errorf("%s arguments not supported", args[1].Type())
+	}
+	length := int(lengthArg.Uint64());
+	if number.BitLen() > length {
+		return runtime.Errorf("%v value requires more than %d bits", number, length)
+	}
+
+	return &runtime.Bitstring{Value: number.Value(), Unit: runtime.Bit, Padding: 4}
```
Comment (5nord) [inline builtins/builtins.go:57]: The hard-coded padding value strikes me odd. Shouldn't it rather be something like this?
```suggestion
	return &runtime.Bitstring{Value: number.Value(), Unit: runtime.Bit, Padding: length-number.BitLen()}
```

File: builtins/builtins.go [hunk 2]
```diff
@@ -31,6 +31,32 @@ func Rnd(args ...runtime.Object) runtime.Object {
...
+	}
+	if number.Sign() == -1 {
+		return runtime.Errorf("%s invalue is less than zero", args[0].Type())
+	}
+
+	lengthArg, lenOk := args[1].(runtime.Int)
+	if !lenOk {
+		return runtime.Errorf("%s arguments not supported", args[1].Type())
+	}
+	length := int(lengthArg.Uint64());
```
[the remark this answers is not in the document]
Comment (UrbanskiDawid) [reply builtins/builtins.go]: i thought that negative numbers was handled by:
> if number.Sign() == -1 {
but sure this sounds good

done

File: builtins/builtins.go (hunk 1 above)
Comment (UrbanskiDawid) [reply builtins/builtins.go]: bug, fixed

File: interpreter/interpreter_test.go [hunk 3]
```diff
@@ -286,6 +287,43 @@ func TestBuiltinFunction(t *testing.T) {
...
+func TestBuiltinFunctionInt2Bit(t *testing.T) {
+	tests := []struct {
+		input    string
+		expected interface{}
+	}{
+		{`int2bit()`,     "wrong number of arguments. got=0, want=2"},
+		{`int2bit(1)`,    "wrong number of arguments. got=1, want=2"},
+		{`int2bit("",0)`, "string arguments not supported"},
+		{`int2bit(0,"")`, "string arguments not supported"},
+		{`int2bit(1,4)`,  &runtime.Bitstring{Value: big.NewInt(1), Unit: runtime.Bit, Padding: 4}},
```
[the remark this answers is not in the document]
Comment (UrbanskiDawid) [reply interpreter/interpreter_test.go]: i had to rewrite this test
> runtime.NewBitstring
returns two values thats problematic in here, so I went with:
>		{`int2bit(0, "")`, runtime.Errorf("string arguments not supported")},
>		{`int2bit(1, 4)`, "'0001'B"},

File: interpreter/interpreter_test.go [hunk 4]
```diff
@@ -286,6 +287,43 @@ func TestBuiltinFunction(t *testing.T) {
...
+		{`int2bit(4,1)`,  "4 value requires more than 1 bits"},
```
[the remark this answers is not in the document]
Comment (UrbanskiDawid) [reply interpreter/interpreter_test.go]: adding all

done

File: runtime/object.go [hunk 5]
```diff
@@ -224,17 +224,18 @@ func NewString(s string) *String {
 type Bitstring struct {
 	Value *big.Int
 	Unit  Unit
+	Padding Unit
```
[the remark this answers is not in the document]
Comment (UrbanskiDawid) [reply runtime/object.go]: true, bad name, changing to Length

File: builtins/builtins.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (5nord) [reply builtins/builtins.go]: Negative numbers have been handled, but I believe the length argument was implicitly converted.

File: runtime/object.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (5nord) [reply runtime/object.go]: The two biggest problems in computer science are: Naming things, Cache invalidation and one-off errors :grin:

File: interpreter/interpreter_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (5nord) [reply interpreter/interpreter_test.go]: Ah, I see! Sorry, this slipped my mind.

A helpful practice in testing is to create a helper function for that, for example:
```
func testBitstring(t *testing.T, s string) *runtime.Bitstring {
    bs, err := runtime.NewBitstring(s)
    if err != nil {
        t.Fatalf("malformed bitstring: %s", err.Error())
    }
    return bs
}

**Comment by 5nord** [comment]
Thank you, Dawid!

**Pull request merged**